### ISSUE TITLE
readNetFromTensorflow getConstBlob error when loading model from Microsoft Custom Vision

### ISSUE THREAD

**Issue by flyahobie** [opencv/opencv#11542]
##### System information (version)

- OpenCV => 3.4.1
- Operating System / Platform =>Windows 64 Bit
- Compiler => Visual Studio 2015

##### Detailed description
I have been working with Microsoft Custom Vision API (http://customvision.ai). This allows users to train a network using transfer learning and export TensorFlow models (using the "Compact Model"). The models produced as protobuffer files can be imported and run directly under TensorFlow. When loading in via cv::dnn::readNetFromTensorflow the following exception is thrown:

ImportTensorFlowOpenCV Error: Unspecified error (More than one input is Const op) in cv::dnn::experimental_dnn_v3::`anonymous-namespace'::TFImporter::getConstBlob, file C:\opencv\sources\modules\dnn\src\tensorflow\tf_importer.cpp, line 571
FAIL - cv::dnn::readNetFromTensorflow D:\tmp\20180517-2-model.pb
 exception info C:\opencv\sources\modules\dnn\src\tensorflow\tf_importer.cpp:571: error: (-2) More than one input is Const op in function cv::dnn::experimental_dnn_v3::`anonymous-namespace'::TFImporter::getConstBlob

Anyone have any guidance for resolving this error or creating a work around?

##### Steps to reproduce
net = cv::dnn::readNetFromTensorflow(modelfilename);
[20180517-2-model.zip](https://github.com/opencv/opencv/files/2017377/20180517-2-model.zip)

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example

-->

**Comment by dkurt** [comment]
@flyahobie, Thank you! Looks like the problem is in new operation [tf.negative](https://www.tensorflow.org/api_docs/python/tf/negative) introduced at TensorFlow 1.8.
At the beginning of graph, input image is normalized:
![screenshot from 2018-05-18 18-16-49](https://user-images.githubusercontent.com/25801568/40242959-bf692f3c-5ac7-11e8-8887-4585dc9c12fc.png)
May I ask you to fuse constant ops in your graph and try to import it again?

```python
import tensorflow as tf
from tensorflow.python.tools import optimize_for_inference_lib
from tensorflow.tools.graph_transforms import TransformGraph

with tf.gfile.FastGFile('20180517-2-model.pb') as f:
    graph_def = tf.GraphDef()
    graph_def.ParseFromString(f.read())
    graph_def = optimize_for_inference_lib.optimize_for_inference(graph_def, ['Placeholder'], ['loss'], tf.float32.as_datatype_enum)
    graph_def = TransformGraph(graph_def, ['Placeholder'], ['loss'], ['fold_constants', 'sort_by_execution_order'])
    with tf.gfile.FastGFile('fused_graph.pb', 'wb') as f:
            f.write(graph_def.SerializeToString())
```
![screenshot from 2018-05-18 18-19-31](https://user-images.githubusercontent.com/25801568/40243114-10e061aa-5ac8-11e8-9125-1d917a14801a.png)

**Comment by flyahobie** [comment]
Thanks Dmitry. I tried your suggestion and ran into a different problem. I realized I was not running 1.8 and while upgrading TF and its dependencies we had some major changes at work. I will try to update you when I can. I just wanted to say thanks.Pete

Sent from Yahoo Mail on Android 
 
  On Fri, May 18, 2018 at 11:21 AM, Dmitry Kurtaev<notifications@github.com> wrote:   
@flyahobie, Thank you! Looks like the problem is in new operation tf.negative introduced at TensorFlow 1.8.
At the beginning of graph, input image is normalized:

May I ask you to fuse constant ops in your graph and try to import it again?
import tensorflow as tf
from tensorflow.python.tools import optimize_for_inference_lib
from tensorflow.tools.graph_transforms import TransformGraph

with tf.gfile.FastGFile('20180517-2-model.pb') as f:
    graph_def = tf.GraphDef()
    graph_def.ParseFromString(f.read())
    graph_def = optimize_for_inference_lib.optimize_for_inference(graph_def, ['Placeholder'], ['loss'], tf.float32.as_datatype_enum)
    graph_def = TransformGraph(graph_def, ['Placeholder'], ['loss'], ['fold_constants', 'sort_by_execution_order'])
    with tf.gfile.FastGFile('fused_graph.pb', 'wb') as f:
            f.write(graph_def.SerializeToString())
